### DISCUSSION TITLE
Add toggles to rules

### DISCUSSION THREAD

**Discussion by NyalephTheCat** [pest-parser/pest#886] [Ideas]
When reading the ecma script specs (https://tc39.es/ecma262/multipage/grammar-summary.html#sec-grammar-summary), you'll find rules  defined like this:

![image](https://github.com/pest-parser/pest/assets/97960996/5369aeb4-787f-49c2-adfd-a32217db0ef6)

Basicly it would be good to be able to write rules in a similar way, where it would effectively create multiples rules, like:

```pest
IdentifierReference = { Identifier }
IdentifierReference_Yield = { Identifier | yield }
IdentifierReference_Await = { Identifier | await }
IdentifierReference_Yield_Await = { Identifier | yield | await }
```

You can read about the ecma script notational conventions for this here: https://tc39.es/ecma262/multipage/notational-conventions.html#sec-grammatical-parameters

**Comment by tomtau** [comment]
So the idea is that a parametrised rule:
```
IdentifierReference[Yield, Await] = { Identifier ~ [~Yield, "yield"] ~ [~Await, "await"] }
```
would expand to:
```
IdentifierReference = { Identifier }
IdentifierReference_Yield = { Identifier | yield }
IdentifierReference_Await = { Identifier | await }
IdentifierReference_Yield_Await = { Identifier | yield | await }
```

Is that correct? It looks cool. You can perhaps post a thread in https://github.com/pest-parser/pest/discussions/885 for more visibility?

**Reply by NyalephTheCat** [reply]
For sure!
